lib/logstash/codecs/json_lines.rb

```diff
@@ -43,12 +52,22 @@ def decode(data, &block)
   end
 
   def encode(event)
+    @encoder.call(event)
+  end
+
+  private
+
+  def encode_default(event)
     # Tack on a @delimiter for now because previously most of logstash's JSON
     # outputs emitted one per line, and whitespace is OK in json.
     @on_event.call(event, "#{event.to_json}#{@delimiter}")
   end
 
-  private
+  def encode_with_metadata(event)
+    # Tack on a @delimiter for now because previously most of logstash's JSON
+    # outputs emitted one per line, and whitespace is OK in json.
+    @on_event.call(event, "#{event.to_json_with_metadata}#{@delimiter}")
```

Comment:
I think you can use `LogStash::Json.dump(event.to_hash_with_metadata)` here instead since `to_json_with_metadata` has gone away.